src/dashboard/components/button.py

```diff
@@ -0,0 +1,38 @@
+"""Button component."""
+from typing import Dict
+
+from dash import html
+
+from dashboard.components.icon import icon
+
+
+def button(
+    icon_name: str,
+    text: str,
+    fill: int = 0,
+    weight: int = 400,
+    grade: int = 0,
+    size: int = 16,
+    _id: str | Dict[str, str] = "",
+) -> html.Button:
```

Comment:
Good idea! Unfortunately kwargs' keys and values can't be typed as it uses the Unpack type which requires >=3.11. (https://stackoverflow.com/a/63550734) And only using TypedDict on the icon_kwargs parameter would require creating a class for the icon component parameters.  Should I just not type icon_kwargs? Extraction of icon_name, fill etc. could then be done explicitly inside the button component using .pop before passing to icon...

Reply:
In this case, `icon_kwargs` could be typed as `dict[str, int]`, and would require treating `icon_name` separately. Having `icon_name` as a positional argument is reasonable in this case, since there is no sensible default value.

Obviously this is not a perfect solution, maybe using TypedDict would be better?

Reply:
Using a TypedDict would be nice, but it is also not possible to set default values in a TypedDict. What could be done is unpacking a TypedDict object with the total class parameter set to False, i.e: 
`class IconKWArgs(TypedDict, total=False)`
This would allow for not having to set all IconKWArgs values for it to pass linting, as well as only setting the icon parameters that were passed to the button icon_kwargs parameter.